### ISSUE TITLE
Cal the HausdorffLoss is toooooo large

### ISSUE THREAD

**Issue by ZhixiangWang-CN** [SilmarilBearer/HausdorffLoss#2]
Hello,
I have tried to use HausdirffDTLoss as Lossfunction.
However, the result for HausdirffDTLoss is too large.
For example:
```
output = np.zeros((3,3,512,512))
output[:,:,100:200,100:200] =1
output[:,:,300:500,300:500] =0.5
output_t = torch.tensor(output)
label = np.zeros((3,3,512,512))
label[:,:,100:200,100:200] =1
label_t = torch.tensor(label)
HD_loss =HausdorffDTLoss()
result = HD_loss(output_t,label_t)
```

the result is 6642.6849.

It is too huge.Did I do something wrong? Or do you have some method to normalize them into 0-1?

Thanks

**Comment by PatRyg99** [comment]
Haussdorf loss is a distance based loss so it doesn't have upper boundary of values it can take - it is solely based on the dataset itself. I don't think anything is wrong here just a way the loss behaves - raising distance fields to power of alpha generates such big values but it's in order with the paper itself. May I ask, did you expect it to be between 0-1 thus your confusion?

**Comment by ZhixiangWang-CN** [comment]
Thank you for your reply.
I want to combine HD Loss and Dice Loss together.
However, the Dice Loss is between 0-1.
So I want to normalize it into 0-1.

**Comment by PatRyg99** [comment]
Problem with normalizing Haussdorf itself is that its max value can change between batches thus simple normalization technic would be batch dependent which it's not correct. Maybe you could try to find max haussdorf value over whole dataset and use it as a normalization term to divide by. Though I don't know how this would behave.

Common idea for using hybrid losses is to introduce some parameter beta:
`loss = (1-beta) * dice + beta * HD`

Selection and usage of such parameter is described in this paper: https://arxiv.org/pdf/1812.07032.pdf. It's about boundary loss but it is also a distance based loss so it should work fine as described in this paper in experiments section https://openreview.net/pdf?id=NDEmtyb4cXu. From my experience starting with bigger parameter for region based losses and decreasing it over time to amplify distance loss term is a way to go, cause distance losses are a bit unstable when predictions are too random, but it may depend on data.

Hope you find those ideas useful in your problem.

**Comment by ZhixiangWang-CN** [comment]
Wow~ your idea is amazing! Thanks for your help.

**Comment by pangda72** [comment]
> Thank you for your reply. I want to combine HD Loss and Dice Loss together. However, the Dice Loss is between 0-1. So I want to normalize it into 0-1.

Hi， @ZhixiangWang-CN 
Have you normalized Hausdorff Loss to 0-1?I want to combine HD Loss and Dice Loss together too,but I don't know how I can solve this problem. Could you give me some help ?

Thanks in advance.
Looking forward to your reply.
Best.
